### DISCUSSION TITLE
Send message to channels - discord.js

### DISCUSSION THREAD

**Discussion by rahultrivedi180** [discordjs/discord.js#5196] [Q&A] [answered]
```javascript
const Discord = require("discord.js");
const client = new Discord.Client();

const token = "token";

client.login(token);

client.channels.cache.get("channel_id").send("Hello!");
```

![](https://user-images.githubusercontent.com/32861078/104095137-38c92380-52bb-11eb-9632-5f4021d01c70.png)

I can not see the send method even in the code suggestion. Channel is not cached maybe. Is there any other way to send messages to a specific channel?

**Comment by Androz2091** [comment, marked as answer]
Hello @rahultrivedi180, you have to wait for the client to be ready and send the message. You can use the following code:
```js
const Discord = require("discord.js");
const client = new Discord.Client();

const token = "token";

client.login(token);

client.on("ready", () => {
    client.channels.cache.get("channel_id").send("Hello!");
});
```
Before the ready event `client.channels.cache.get("channel_id")` is undefined because the channel is not cached.

**Reply by Androz2091** [reply]
@rahultrivedi180 please use the "mark as answer" button to close this issue
